pandas/tests/io/formats/test_to_html.py

```diff
@@ -1881,3 +1881,65 @@ def test_to_html_with_index_names_false(self):
                                                         name='myindexname'))
         result = df.to_html(index_names=False)
         assert 'myindexname' not in result
+
+    def test_to_html_styler_basic(self):
+        df = pd.DataFrame({"A": [-1, 1], "B": [-2, 2]}).style
+        df.uuid = 42
+        expected = df.render()
+        actual = df.to_html()
+        assert actual == expected
+
+    def test_to_html_styler_file(self):
+        import os
```

Comment:
Have a look at `pandas.util.testing.ensure_clean`. So it'd be something like

```python
with tm.ensure_clean() as path:
    df.to_html(path)
    with open(path) as p:
        result = p.read()
```

Then you don't have to worry about making / cleaning up files yourself.

Reply:
Thanks, I'll look into that